editor/effects.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+import { parse, stringify } from 'querystring';
+
+/**
+ * WordPress dependencies
+ */
+import { getBlockSettings, switchToBlockType } from 'blocks';
+
+/**
+ * Internal dependencies
+ */
+import { focusBlock, replaceBlocks } from './actions';
+
+export default {
+	REQUEST_POST_UPDATE( action, store ) {
+		const { dispatch } = store;
+		const { postId, edits } = action;
+		const isNew = ! postId;
+		const toSend = postId ? { id: postId, ...edits } : edits;
+
+		new wp.api.models.Post( toSend ).save().done( ( newPost ) => {
+			dispatch( {
+				type: 'REQUEST_POST_UPDATE_SUCCESS',
+				post: newPost,
+			} );
+		} ).fail( ( err ) => {
+			dispatch( {
+				type: 'REQUEST_POST_UPDATE_FAILURE',
+				error: get( err, 'responseJSON', {
```

Comment:
>That can serve multipurpose cases rather than just network. These are often triggered by the success/failure network actions.

Yep, this is exactly the sort of thing I'd hope for.